src/CSI.IBTA.BenefitsService/Services/UserBalanceService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using CSI.IBTA.BenefitsService.Interfaces;
+using CSI.IBTA.DataLayer.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using CSI.IBTA.Shared.Entities;
+using CSI.IBTA.Shared.DTOs;
+using CSI.IBTA.Shared.DTOs.Errors;
+using System.Net;
+
+namespace CSI.IBTA.BenefitsService.Services
+{
+    public class UserBalanceService : IUserBalanceService
+    {
+        private readonly IBenefitsUnitOfWork _benefitsUnitOfWork;
+
+        public UserBalanceService(IBenefitsUnitOfWork benefitsUnitOfWork)
+        {
+            _benefitsUnitOfWork = benefitsUnitOfWork;
+        }
+
+        public async Task<GenericResponse<decimal>> GetCurrentBalanceForPlan(int planId)
+        {
+            var plan = await _benefitsUnitOfWork.Plans
+                .Include(c => c.Package)
+                .FirstOrDefaultAsync(x => x.Id == planId);
+
+            if(plan == null) return new GenericResponse<decimal>(HttpErrors.ResourceNotFound, 0);
+            if(!plan.Package.IsActive) return new GenericResponse<decimal>(new HttpError("This plan is not active yet", HttpStatusCode.BadRequest), 0);
+
+            var package = plan.Package;
+            var negativeTransactions = await _benefitsUnitOfWork.Claims
+                .Find(x => x.PlanId == plan.Id && x.Status == ClaimStatus.Approved);
+
+            var totalPeriods = 0;
+            int periodsPassed = 0;
+            switch (package.PayrollFrequency) 
+            {
+                case PayrollFrequency.Weekly:
+                    totalPeriods += (int)Math.Ceiling((package.PlanEnd - package.PlanStart).TotalDays / 7.0);
+
+                    periodsPassed += (int)Math.Ceiling((DateTime.Now - package.PlanStart).TotalDays / 7.0);
```

Comment:
Inconsistency: DateTime.Now and DateTime.Utc... are used for calculations.
UTC may show different day and even year than current time. 
Let say, in UTC the date is "1999-12-31 23:00:00" 
If timezone is GMT+2, than DateTime.Now would be "2000-01-01 01:00:00".

So, If PlanStart was set in current time ( DateTime.Now), then use DateTime.**Today** as a baseline. 
If PlanStart has been created and adjusted to UTC, then  use DateTime.**UtcToday** as a baseline.